amd/src/courseformat/courseindex/section.js

```diff
@@ -119,23 +146,36 @@ export default class Component extends ComponentBase {
     hideDropZone(dropdata) {
         if (dropdata.type == 'cm') {
             this.getLastCm()?.classList.remove(this.classes.DROPDOWN);
-        } else {
+        }
+        if (dropdata.type == 'section') {
             const state = this.reactive.stateManager.state;
             const origin = state.section.get(dropdata.id);
             let target = this.section;
             while (target.levelsan > origin.levelsan) {
                 target = state.section.get(this.course.sectionlist[target.number - 1]);
             }
             const moveDirection = Math.sign(target.number - origin.number);
-            let targetEnd = target;
-            while (this.course.sectionlist.length > targetEnd.number + 1
-                    && state.section.get(this.course.sectionlist[targetEnd.number + 1]).levelsan > origin.levelsan) {
-                targetEnd = state.section.get(this.course.sectionlist[targetEnd.number + 1]);
+            let targetShow = target;
+            let targetShowBorder = moveDirection;
+            if (moveDirection > 0 && !target.indexcollapsed) {
+                let targetChild = target;
+                while (this.course.sectionlist.length > targetChild.number + 1
+                && state.section.get(this.course.sectionlist[targetChild.number + 1]).levelsan > target.levelsan) {
+                    targetChild = state.section.get(this.course.sectionlist[targetChild.number + 1]);
+                    if (targetChild.levelsan <= origin.levelsan) {
+                        targetShow = targetChild;
+                        targetShowBorder = -1;
+                        break;
+                    }
+                }
             }
             const targetHTML = document.querySelector(
-                ".courseindex-section[data-id='" + (moveDirection <= 0 ? target : targetEnd).id + "']");
-            targetHTML.classList.remove(this.classes.DROPUP);
-            targetHTML.classList.remove(this.classes.DROPDOWN);
+                ".courseindex-section[data-id='" + targetShow.id + "']");
+            if (targetShowBorder > 0) {
+                targetHTML.classList.remove(this.classes.DROPDOWN);
+            } else {
```

Comment:
Again, need to ignore if a section is dragged over itself.
```suggestion
            } else if (targetShowBorder < 0) {
```